### PR TITLE
update type select to fundamental-react 0.13

### PR DESCRIPTION
**Pull request by auryn31**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://sap.github.io/fundamental-react/?path=/docs/component-api-select--primary
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/fundamental-react/lib/Select/Select.d.ts [hunk 1]
```diff
@@ -1,21 +1,33 @@
...
+    textContentClassName?: string;
+    triggerClassName?: string;
     validationState?: {
         state?: 'error' | 'warning' | 'information' | 'success';
         text?: string;
     };
+    options?: Array<{ key: string; text: string }>;
     onClick?: (...args: any[]) => any;
+    onBlur?: (...args: any[]) => any;
+    onSelect?: (...args: any[]) => any;
```
Comment (skvale) [inline types/fundamental-react/lib/Select/Select.d.ts:26]: A few suggested changes
```ts
export type Option = { key: string; text: string; };
// ...
    options?: Option[];
// ...
onBlur?: (event: React.FocusEvent<HTMLDivElement>) => void;
onSelect?: (event: React.MouseEvent<HTMLLIElement> | React.KeyboardEvent<HTMLLIElement>, selectedOption: Option) => void

File: types/fundamental-react/lib/Select/Select.d.ts [hunk 2]
```diff
@@ -1,21 +1,33 @@
-import * as React from "react";
+import * as React from 'react';
 
 export interface SelectProps {
```
Comment (skvale) [inline types/fundamental-react/lib/Select/Select.d.ts:3]: Additional props
```ts
popoverProps?: Record<string, any>;
validationOverlayProps?: {
  className?: string;
  formMessageProps?: Record<string, any>;
  innerRefClassName?: string;
  popperClassName?: string;
  referenceClassName?: string;
  wrapperProps?: Record<string, any>;
}
```

Comment (auryn31) [reply types/fundamental-react/lib/Select/Select.d.ts:3]: thanks for the addons, i will add them

File: types/fundamental-react/lib/Select/Select.d.ts (hunk 1 above)
Comment (auryn31) [reply types/fundamental-react/lib/Select/Select.d.ts:26]: absolutely right, i update the pr

**Review by milanpro — APPROVED** [review]
Well done :+1:

**Comment by auryn31** [comment]
Ready to merge

File: types/fundamental-react/lib/Select/Select.d.ts (hunk 2 above)
Comment (skvale) [reply types/fundamental-react/lib/Select/Select.d.ts:4]: Did I miss something?  These should have been included right?

popoverProps: https://github.com/SAP/fundamental-react/blob/master/src/Select/Select.js#L258

validationOverlayProps: https://github.com/SAP/fundamental-react/blob/master/src/Select/Select.js#L268

File: types/fundamental-react/lib/Select/Select.d.ts [hunk 3]
```diff
@@ -1,17 +1,35 @@
...
+    selectedKey?: string;
+    textContentClassName?: string;
+    triggerClassName?: string;
     validationState?: {
         state?: 'error' | 'warning' | 'information' | 'success';
         text?: string;
     };
-    onClick?: (...args: any[]) => any;
+    options?: Option[];
+    onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
```
Comment (skvale) [inline types/fundamental-react/lib/Select/Select.d.ts:30]: I think the onClick ends up on a div for Select
```ts
onClick?: (event: React.MouseEvent<HTMLDivElement>) => void;
```

**Pull request merged**